### PR TITLE
Filter by published state

### PR DESCRIPTION
**Pull request by cavis**
For #505.  Adds a new `state` filter to authorized story endpoints.  Filters include (url-encoding the `=`):

- [x] `?filters=state%3Dpublished` (published_at in the past)
- [x] `?filters=state%3Dscheduled` (published_at in the future)
- [x] `?filters=state%3Ddraft` (published_at is null)

Endpoints include:

- [x] `/api/v1/authorization/stories`
- [x] `/api/v1/authorization/accounts/:account_id/stories`
- [x] `/api/v1/authorization/networks/:network_id/stories`
- [x] `/api/v1/authorization/series/:series_id/stories`

And also added to the search endpoints:

- [x] `/api/v1/authorization/stories/search`
- [x] `/api/v1/authorization/series/:series_id/stories/search`

### CODE REVIEW

File: app/controllers/api/auth/stories_controller.rb [hunk 1]
```diff
@@ -8,7 +8,7 @@ class Api::Auth::StoriesController < Api::StoriesController
 
   filter_resources_by :account_id, :series_id, :network_id
 
-  filter_params :highlighted, :purchased, :v4, :text, :noseries
+  filter_params :highlighted, :purchased, :v4, :text, :noseries, :state
```
Comment (cavis) [inline app/controllers/api/auth/stories_controller.rb]: I opted to call this `state` and give it a single value.  The alternative would be completely separate filters like:

```
?filters=published
?filters=scheduled
?filters=draft
```

Open to suggestions on which is better.

File: docker-compose-ci.yml [hunk 2]
```diff
@@ -43,7 +43,7 @@ services:
     volumes:
       - ./test/db/:/docker-entrypoint-initdb.d/
   elasticsearch:
-    image: docker.elastic.co/elasticsearch/elasticsearch-oss:6.3.0
+    image: elasticsearch:6.4.3
```
Comment (cavis) [inline docker-compose-ci.yml]: Was having issues with that `elastic.co` repo.  So switched to the official dockerhub instance, which is supposedly the same, but not hosted by Elastic.  Also upgraded to 6.4, which is what prod-cms uses right now.

File: app/controllers/api/auth/stories_controller.rb [hunk 3]
```diff
@@ -68,6 +81,19 @@ def search
   def search_params
     sparams = super
     sparams[:fq]['series_id'] = 'NULL' if filters.noseries?
+    sparams[:fq]['published_at'] = search_by_state(filters.state) if filters.state?
     sparams
   end
+
+  def search_by_state(state)
+    if state == 'published'
+      '[* TO now]'
```
Comment (cavis) [inline app/controllers/api/auth/stories_controller.rb]: The inclusive/exclusive range syntax is documented [here](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-query-string-query.html#_ranges).

File: app/models/story.rb [hunk 4]
```diff
@@ -122,7 +122,8 @@ def self.paranoia_scope
       select('pieces.*, series.subscription_approval_status, series.subscriber_only_at')
   }
   scope :published, -> { where('`published_at` <= ?', Time.now) }
-  scope :unpublished, -> { where('`published_at` IS NULL OR `published_at` > ?', Time.now) }
+  scope :scheduled, -> { where('`published_at` > ?', Time.now) }
+  scope :draft, -> { where('`published_at` IS NULL', Time.now) }
```
Comment (svevang) [inline app/models/story.rb]: unneeded func param?

Comment (cavis) [reply app/models/story.rb]: ah, good call.

**Pull request merged**